app/controllers/key_pairs_controller.rb

```diff
@@ -0,0 +1,56 @@
+class KeyPairsController < ApplicationController
+  before_action :find_compute_resource, :except => [:download]
+  before_action :find_resource, :only => [:download]
+
+  def index
+    @key_pairs = @compute_resource.get_compute_key_pairs
```

Comment:
missing exception handling for unable to fetch key pairs. can easily be simulated by using the wrong user/pass on your compute resource.